site/app/controllers/forum/ForumController.php

```diff
@@ -527,6 +531,49 @@ public function publishPost() {
                 $result['next_page'] = $this->core->buildCourseUrl(['forum', 'threads', $thread_id]) . '?' . http_build_query(['option' => $display_option]);
                 $result['post_id'] = $post_id;
                 $result['thread_id'] = $thread_id;
+                $posts = $this->core->getQueries()->getPostsInThreads([$result['thread_id']]);
+                $first = true;
+                $first_post_id = 1;
+                $order_array = [];
+                $reply_level_array = [];
+                if ($display_option != 'tree') {
+                    $reply_level = 1;
+                }
+                else {
+                    foreach ($posts as $post) {
+                        if ($thread_id === -1) {
```

Comment:
what type is $thread_id? It looks like here you are comparing a string to an integer which would be incorrect